### ISSUE TITLE
Open Modal with Service

### ISSUE THREAD

**Issue by simkepal** [vladotesanovic/ngSemantic#92]
Hello,

I have a problem. I would like to have an AuthService with function to open a modal. Is this possible? How can this be done?

Thanks for help!

Simon

**Comment by nimaen** [comment]
Hi,

I suppose you want to open your modal from the service.
I suggest you use a route guard instead, then redirect to an appropriate page.

**Comment by simkepal** [comment]
Yes, I want to open modal from service. But I don't want to redirect the user. I want to open a modal with sign in form and if everything is OK than I pass the user to next page.

**Comment by nimaen** [comment]
Well, you could use auxiliary routes.
This way you can open a modal from any route with your service just by redirecting the user to the aux route url.

Have a look here : https://angular-2-training-book.rangle.io/handout/routing/aux-routes.html

Honestly, I think it might be the best approach.
Else, you would want to use some nasty ways like direct dom manipulations, which is not intended.
